pkg/agent/cniserver/pod_monitor.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cniserver
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+	"time"
+
+	"antrea.io/libOpenflow/openflow15"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/openflow"
+	agenttypes "antrea.io/antrea/pkg/agent/types"
+	"antrea.io/antrea/pkg/util/channel"
+)
+
+const (
+	podNotReadyTimeInSeconds = 30 * time.Second
+	// How long to wait before retrying the processing of an Egress change.
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+)
+
+var (
+	// retryInterval is the interval to re-install Pod OpenFlow entries if any error happened.
+	// Note, using a variable rather than constant for retryInterval because we may use a shorter time in the
+	// test code.
+	retryInterval = time.Second * 5
+)
+
+type unreadyPodInfo struct {
+	podName        string
+	podNamespace   string
+	flowsInstalled bool
+	createTime     time.Time
+}
+
+type podIfaceMonitor struct {
+	podLister         corelisters.PodLister
+	kubeClient        clientset.Interface
+	eventBroadcaster  record.EventBroadcaster
+	record            record.EventRecorder
+	ifaceStore        interfacestore.InterfaceStore
+	ofClient          openflow.Client
+	podUpdateNotifier channel.Notifier
+
+	// unreadyInterfaces is a map to store the OVS ports which is waiting for the PortStatus from OpenFlow switch.
+	// The key in the map is the OVS port name, and its value is unreadyPodInfo.
+	// It is used only on Windows now.
+	unreadyInterfaces map[string]*unreadyPodInfo
+	// unreadyMapMutex is a mutex to guard map unreadyInterfaces.
+	unreadyMapMutex sync.Mutex
+	statusCh        chan *openflow15.PortStatus
+
+	queue workqueue.TypedRateLimitingInterface[string]
+	// statusMapMtx is a mutex to guard ovsPortStatusMap
+	statusMapMtx     sync.Mutex
```

Comment:
spell out "Mutex", it's more readable and also consistent with `unreadyMapMutex`